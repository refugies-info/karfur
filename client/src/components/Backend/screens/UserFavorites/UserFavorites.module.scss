@import "~scss/mixins/unit";
@import "~scss/colors";

.container {
  background: $gray40b;
  border-radius: 12px;
  padding-top: u(10);
  padding-bottom: u(10);
  width: 1120px;
  height: fit-content;
  margin: u(6) auto;
}

.cards {
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  margin-right: u(8);
  margin-left: u(8);
  gap: u(4);
}

.title {
  margin-right: u(10);
  margin-left: u(10);
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
