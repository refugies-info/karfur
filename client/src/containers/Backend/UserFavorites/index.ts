import UserFavorites from "./UserFavorites";

export default UserFavorites;
