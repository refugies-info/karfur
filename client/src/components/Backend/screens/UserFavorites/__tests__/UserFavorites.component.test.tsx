//@ts-nocheck
import mockAxios from "jest-mock-axios";
import routerMock from "next/router";
import {
  fetchUserFavoritesActionCreator,
  updateUserFavoritesActionCreator,
} from "services/UserFavoritesInLocale/UserFavoritesInLocale.actions";
import { initialMockStore } from "__fixtures__/reduxStore";
import { wrapWithProvidersAndRenderForTesting } from "../../../../../../jest/lib/wrapWithProvidersAndRender";
import UserFavorites from "../UserFavorites";

jest.mock("next/router", () => require("next-router-mock"));

import "jest-styled-components";

jest.mock("services/UserFavoritesInLocale/UserFavoritesInLocale.actions", () => {
  const actions = jest.requireActual("services/UserFavoritesInLocale/UserFavoritesInLocale.actions");

  return {
    updateUserFavoritesActionCreator: jest.fn(actions.updateUserFavoritesActionCreator),
    fetchUserFavoritesActionCreator: jest.fn(actions.fetchUserFavoritesActionCreator),
  };
});

describe("UserFavorites", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockAxios.reset();
  });
  it("should render correctly when loading", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserFavorites,
      reduxState: {
        ...initialMockStore,
        loadingStatus: { FETCH_USER_FAVORITES: { isLoading: true } },
      },
      compProps: { t: (_: string, element2: string) => element2 },
    });
    expect(fetchUserFavoritesActionCreator).toHaveBeenCalledWith("fr");
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly when 0 favorites", () => {
    window.scrollTo = jest.fn();

    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserFavorites,
      compProps: { t: (_: string, element2: string) => element2 },
    });

    expect(fetchUserFavoritesActionCreator).toHaveBeenCalledWith("fr");

    expect(asFragment()).toMatchSnapshot();
  });

  const fav1 = {
    _id: "id1",
    titreInformatif: "tinfo1",
    titreMarque: "titreMarque1",
    abstract: "abstract1",
    typeContenu: "dispositif",
    contenu: [
      {},
      {
        children: [
          {
            title: "Zone d'action",
          },
        ],
      },
    ],
    mainSponsor: {
      picture: { secure_url: "" },
    },
    theme: "6319f6b363ab2bbb162d7df5",
  };

  const fav2 = {
    _id: "id2",
    titreInformatif: "tinfo2",
    titreMarque: "titreMarque2",
    abstract: "abstract2",
    typeContenu: "dispositif",
    contenu: [
      {},
      {
        children: [
          {
            title: "Zone d'action",
          },
        ],
      },
    ],
    mainSponsor: {
      picture: { secure_url: "" },
    },
    theme: "6319f6b363ab2bbb162d7df6",
  };

  const fav3 = {
    _id: "id3",
    titreInformatif: "tinfo3",
    titreMarque: "titreMarque3",
    abstract: "abstract3",
    typeContenu: "demarche",
    contenu: [
      {},
      {
        children: [
          {
            title: "Zone d'action",
          },
        ],
      },
    ],
    mainSponsor: {
      picture: { secure_url: "" },
    },
    theme: "6319f6b363ab2bbb162d7df7",
  };
  it("should render correctly when 3 favorites", () => {
    window.scrollTo = jest.fn();
    const { asFragment } = wrapWithProvidersAndRenderForTesting({
      Component: UserFavorites,
      compProps: { t: (_: string, element2: string) => element2 },
      reduxState: { ...initialMockStore, userFavorites: { favorites: [fav1, fav2, fav3] } },
    });
    expect(fetchUserFavoritesActionCreator).toHaveBeenCalledWith("fr");
    expect(asFragment()).toMatchSnapshot();
  });

  it("should dispatch updateUserFavoritesActionCreator when click on Tout supprimer", () => {
    window.scrollTo = jest.fn();
    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserFavorites,
      compProps: { t: (_: string, element2: string) => element2 },
      reduxState: { ...initialMockStore, userFavorites: { favorites: [fav1, fav2, fav3] } },
    });
    expect(fetchUserFavoritesActionCreator).toHaveBeenCalledWith("fr");
    component.root.findByProps({ "data-testid": "test-delete-button" }).props.onClick();

    expect(updateUserFavoritesActionCreator).toHaveBeenCalledWith({
      type: "remove-all",
      locale: "fr",
    });
  });

  it("should dispatch updateUserFavoritesActionCreator when click on Tout supprimer and language is en", () => {
    routerMock.locale = "en";
    window.scrollTo = jest.fn();

    const component = wrapWithProvidersAndRenderForTesting({
      Component: UserFavorites,
      compProps: { t: (_: string, element2: string) => element2 },
      reduxState: { ...initialMockStore, userFavorites: { favorites: [fav1, fav2, fav3] } },
    });

    expect(fetchUserFavoritesActionCreator).toHaveBeenCalledWith("en");
    component.root.findByProps({ "data-testid": "test-delete-button" }).props.onClick();

    expect(updateUserFavoritesActionCreator).toHaveBeenCalledWith({
      type: "remove-all",
      locale: "en",
    });
  });
});
