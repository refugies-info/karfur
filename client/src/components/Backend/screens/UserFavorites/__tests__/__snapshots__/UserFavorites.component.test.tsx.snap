// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFavorites should render correctly when 0 favorites 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  margin-bottom: 32px;
  align-items: flex-end;
}

.c2 {
  background: #212121;
  border-radius: 12px;
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  color: #ffffff;
  padding: 8px 19px 8px 19px;
  margin: 0px 8px 0px 8px;
}

.c1 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c3 {
  background: #ffffff;
  border-radius: 12px;
  padding: 13px;
  width: 248px;
  height: 248px;
  font-weight: bold;
  font-size: 22px;
  line-height: 28px;
  color: #2d9cdb;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.c4 {
  background: #ffffff;
  border-radius: 12px;
  width: 248px;
  height: 248px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.c5 {
  background: #c6c6c6;
  border-radius: 8px;
  width: 221px;
  height: 28px;
  margin-bottom: 8px;
}

.c6 {
  background: #c6c6c6;
  border-radius: 8px;
  width: 130px;
  height: 28px;
  margin-bottom: 8px;
}

.c7 {
  background: #c6c6c6;
  border-radius: 3px;
  width: 221px;
  height: 14px;
  margin-bottom: 5px;
  margin-left: 13px;
}

.c8 {
  background: #c6c6c6;
  border-radius: 3px;
  width: 200px;
  height: 14px;
  margin-bottom: 5px;
  margin-left: 13px;
}

.c9 {
  background: #c6c6c6;
  border-radius: 3px;
  width: 180px;
  height: 14px;
  margin-bottom: 5px;
  margin-left: 13px;
}

.c10 {
  background: #c6c6c6;
  height: 50px;
  width: 100%;
  border-radius: 0px 0px 12px 12px;
  margin-top: 16px;
  padding-top: 18px;
  padding-left: 50px;
}

.c11 {
  background: #ffffff;
  border-radius: 3px;
  width: 180px;
  height: 14px;
}

<div
  className="container"
>
  <div
    style={
      {
        "marginLeft": "40px",
        "marginRight": "40px",
      }
    }
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous avez
      </div>
      <div
        className="c2"
      >
        0
      </div>
      <div
        className="c1"
      >
        UserFavorites.content_saved
      </div>
    </div>
  </div>
  <div
    className="cards"
  >
    <div
      className="c3"
    >
      UserFavorites.Comment ajouter
      <button
        className="btn tuto me-2"
        onClick={[Function]}
      >
        <span
          className="me-2 icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-question-mark-circle-outline" fill="#fff"><g data-name="Layer 2"><g data-name="menu-arrow-circle"><rect width="24" height="24" transform="rotate(180 12 12)" opacity="0"/><path d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"/><path d="M12 6a3.5 3.5 0 0 0-3.5 3.5 1 1 0 0 0 2 0A1.5 1.5 0 1 1 12 11a1 1 0 0 0-1 1v2a1 1 0 0 0 2 0v-1.16A3.49 3.49 0 0 0 12 6z"/><circle cx="12" cy="17" r="1"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        UserFavorites.tutoriel
      </button>
    </div>
    <div
      className="c4"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <div
          className="c5"
          width="221px"
        />
        <div
          className="c6"
          width="130px"
        />
      </div>
      <div>
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c8"
          width="200px"
        />
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c9"
          width="180px"
        />
        <div
          className="c10"
        >
          <div
            className="c11"
          />
        </div>
      </div>
    </div>
    <div
      className="c4"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <div
          className="c5"
          width="221px"
        />
        <div
          className="c6"
          width="130px"
        />
      </div>
      <div>
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c8"
          width="200px"
        />
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c9"
          width="180px"
        />
        <div
          className="c10"
        >
          <div
            className="c11"
          />
        </div>
      </div>
    </div>
    <div
      className="c4"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <div
          className="c5"
          width="221px"
        />
        <div
          className="c6"
          width="130px"
        />
      </div>
      <div>
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c8"
          width="200px"
        />
        <div
          className="c7"
          width="221px"
        />
        <div
          className="c9"
          width="180px"
        />
        <div
          className="c10"
        >
          <div
            className="c11"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserFavorites should render correctly when 3 favorites 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  margin-bottom: 32px;
  align-items: flex-end;
}

.c2 {
  background: #212121;
  border-radius: 12px;
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  color: #ffffff;
  padding: 8px 19px 8px 19px;
  margin: 0px 8px 0px 8px;
}

.c1 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c5 :hover,
.favorite:hover+.c5 {
  background-color: #DDF2F9!important;
  border-color: #06508C!important;
  color: #06508C!important;
}

.c3 :hover,
.favorite:hover+.c3 {
  background-color: #EFE8F4!important;
  border-color: #3D2884!important;
  color: #3D2884!important;
}

.c4 :hover,
.favorite:hover+.c4 {
  background-color: #EEE7E5!important;
  border-color: #443023!important;
  color: #443023!important;
}

<div
  className="container"
>
  <div
    className="title"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous avez
      </div>
      <div
        className="c2"
      >
        3
      </div>
      <div
        className="c1"
      >
        UserFavorites.contents_saved
      </div>
    </div>
    <div>
      <button
        className="btn outline-black"
        data-test-id="test-delete-button"
        onClick={[Function]}
      >
        <span
          className="me-2 icon"
        >
          <i
            dangerouslySetInnerHTML={
              {
                "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-trash-outline" fill="#fff"><g data-name="Layer 2"><g data-name="trash"><rect width="24" height="24" opacity="0"/><path d="M21 6h-5V4.33A2.42 2.42 0 0 0 13.5 2h-3A2.42 2.42 0 0 0 8 4.33V6H3a1 1 0 0 0 0 2h1v11a3 3 0 0 0 3 3h10a3 3 0 0 0 3-3V8h1a1 1 0 0 0 0-2zM10 4.33c0-.16.21-.33.5-.33h3c.29 0 .5.17.5.33V6h-4zM18 19a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V8h12z"/></g></g></svg>",
              }
            }
            style={
              {
                "height": "20px",
                "lineHeight": "20px",
              }
            }
          />
        </span>
        UserFavorites.Tout supprimer
      </button>
    </div>
  </div>
  <div
    className="cards"
  >
    <div>
      <div
        className="wrapper wrapper"
      >
        <button
          className="btn favorite btn btn-secondary"
          onClick={[Function]}
          title="Dispositif.addToFavorites"
          type="button"
        >
          <span
            className="icon_outline icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star-outline" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18zM12 16.1a.92.92 0 0 1 .46.11l3.77 2-.72-4.21a1 1 0 0 1 .29-.89l3-2.93-4.2-.62a1 1 0 0 1-.71-.56L12 5.25 10.11 9a1 1 0 0 1-.75.54l-4.2.62 3 2.93a1 1 0 0 1 .29.89l-.72 4.16 3.77-2a.92.92 0 0 1 .5-.04z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          <span
            className="icon_fill icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
        <a
          className="c3 card dispositif content card"
          href="/program/id1"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          title="tinfo1"
        >
          <div
            className="location"
          >
            <img
              alt="Infocards.location"
              data-nimg="1"
              decoding="async"
              height={16}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={16}
            />
            <span
              className="ms-1"
              style={
                {
                  "color": "#3D2884",
                }
              }
            />
          </div>
          <h3
            className="title"
            dangerouslySetInnerHTML={
              {
                "__html": "tinfo1",
              }
            }
            style={
              {
                "color": "#3D2884",
              }
            }
          />
          <div
            className="text max_lines abstract"
            dangerouslySetInnerHTML={
              {
                "__html": "abstract1",
              }
            }
            style={
              {
                "color": "#3D2884",
              }
            }
          />
          <div
            className="infos text my-3"
            style={
              {
                "color": "#3D2884",
              }
            }
          />
          <div
            className="themes"
          >
            <div
              className="container badges theme-badge"
              style={
                {
                  "backgroundColor": "#EFE8F4",
                }
              }
            >
              <span
                className="icon"
              >
                <span
                  className="icon theme-icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": "<svg width="12" height="12"></svg>",
                    }
                  }
                  style={
                    {
                      "height": 12,
                      "width": 12,
                    }
                  }
                />
              </span>
              <span
                style={
                  {
                    "color": "#3D2884",
                  }
                }
              >
                Français
              </span>
            </div>
          </div>
          <div
            className="sponsor"
            style={
              {
                "borderColor": "#3D2884",
              }
            }
          >
            <span
              className="picture"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={40}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style={
                  {
                    "color": "transparent",
                    "objectFit": "contain",
                  }
                }
                width={40}
              />
            </span>
            <span
              className="text max_lines ms-2"
              dangerouslySetInnerHTML={
                {
                  "__html": "titreMarque1",
                }
              }
              style={
                {
                  "color": "#3D2884",
                }
              }
            />
          </div>
        </a>
      </div>
    </div>
    <div>
      <div
        className="wrapper wrapper"
      >
        <button
          className="btn favorite btn btn-secondary"
          onClick={[Function]}
          title="Dispositif.addToFavorites"
          type="button"
        >
          <span
            className="icon_outline icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star-outline" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18zM12 16.1a.92.92 0 0 1 .46.11l3.77 2-.72-4.21a1 1 0 0 1 .29-.89l3-2.93-4.2-.62a1 1 0 0 1-.71-.56L12 5.25 10.11 9a1 1 0 0 1-.75.54l-4.2.62 3 2.93a1 1 0 0 1 .29.89l-.72 4.16 3.77-2a.92.92 0 0 1 .5-.04z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          <span
            className="icon_fill icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
        <a
          className="c4 card dispositif content card"
          href="/program/id2"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          title="tinfo2"
        >
          <div
            className="location"
          >
            <img
              alt="Infocards.location"
              data-nimg="1"
              decoding="async"
              height={16}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={16}
            />
            <span
              className="ms-1"
              style={
                {
                  "color": "#443023",
                }
              }
            />
          </div>
          <h3
            className="title"
            dangerouslySetInnerHTML={
              {
                "__html": "tinfo2",
              }
            }
            style={
              {
                "color": "#443023",
              }
            }
          />
          <div
            className="text max_lines abstract"
            dangerouslySetInnerHTML={
              {
                "__html": "abstract2",
              }
            }
            style={
              {
                "color": "#443023",
              }
            }
          />
          <div
            className="infos text my-3"
            style={
              {
                "color": "#443023",
              }
            }
          />
          <div
            className="themes"
          >
            <div
              className="container badges theme-badge"
              style={
                {
                  "backgroundColor": "#EEE7E5",
                }
              }
            >
              <span
                className="icon"
              >
                <span
                  className="icon theme-icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": "<svg width="12" height="12"></svg>",
                    }
                  }
                  style={
                    {
                      "height": 12,
                      "width": 12,
                    }
                  }
                />
              </span>
              <span
                style={
                  {
                    "color": "#443023",
                  }
                }
              >
                Administratif
              </span>
            </div>
          </div>
          <div
            className="sponsor"
            style={
              {
                "borderColor": "#443023",
              }
            }
          >
            <span
              className="picture"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={40}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style={
                  {
                    "color": "transparent",
                    "objectFit": "contain",
                  }
                }
                width={40}
              />
            </span>
            <span
              className="text max_lines ms-2"
              dangerouslySetInnerHTML={
                {
                  "__html": "titreMarque2",
                }
              }
              style={
                {
                  "color": "#443023",
                }
              }
            />
          </div>
        </a>
      </div>
    </div>
    <div>
      <div
        className="wrapper"
      >
        <button
          className="btn favorite btn btn-secondary"
          onClick={[Function]}
          title="Dispositif.addToFavorites"
          type="button"
        >
          <span
            className="icon_outline icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star-outline" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18zM12 16.1a.92.92 0 0 1 .46.11l3.77 2-.72-4.21a1 1 0 0 1 .29-.89l3-2.93-4.2-.62a1 1 0 0 1-.71-.56L12 5.25 10.11 9a1 1 0 0 1-.75.54l-4.2.62 3 2.93a1 1 0 0 1 .29.89l-.72 4.16 3.77-2a.92.92 0 0 1 .5-.04z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
          <span
            className="icon_fill icon"
          >
            <i
              dangerouslySetInnerHTML={
                {
                  "__html": "<svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="eva eva-star" fill="dark"><g data-name="Layer 2"><g data-name="star"><rect width="24" height="24" transform="rotate(90 12 12)" opacity="0"/><path d="M17.56 21a1 1 0 0 1-.46-.11L12 18.22l-5.1 2.67a1 1 0 0 1-1.45-1.06l1-5.63-4.12-4a1 1 0 0 1-.25-1 1 1 0 0 1 .81-.68l5.7-.83 2.51-5.13a1 1 0 0 1 1.8 0l2.54 5.12 5.7.83a1 1 0 0 1 .81.68 1 1 0 0 1-.25 1l-4.12 4 1 5.63a1 1 0 0 1-.4 1 1 1 0 0 1-.62.18z"/></g></g></svg>",
                }
              }
              style={
                {
                  "height": "20px",
                  "lineHeight": "20px",
                }
              }
            />
          </span>
        </button>
        <a
          className="c5 card demarche content"
          href="/procedure/id3"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          title="tinfo3"
        >
          <div>
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height={48}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={48}
            />
          </div>
          <h3
            className="title"
            dangerouslySetInnerHTML={
              {
                "__html": "tinfo3",
              }
            }
            style={
              {
                "color": "#06508C",
              }
            }
          />
          <div
            className="themes"
          >
            <div
              className="container badges theme-badge"
              style={
                {
                  "backgroundColor": "#DDF2F9",
                }
              }
            >
              <span
                className="icon"
              >
                <span
                  className="icon theme-icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": "<svg width="12" height="12"></svg>",
                    }
                  }
                  style={
                    {
                      "height": 12,
                      "width": 12,
                    }
                  }
                />
              </span>
              <span
                style={
                  {
                    "color": "#06508C",
                  }
                }
              >
                Logement
              </span>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserFavorites should render correctly when loading 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  margin-bottom: 32px;
  align-items: flex-end;
}

.c2 {
  background: #212121;
  border-radius: 12px;
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  color: #ffffff;
  padding: 8px 19px 8px 19px;
  margin: 0px 8px 0px 8px;
}

.c1 {
  font-weight: bold;
  font-size: 28px;
  line-height: 35px;
  margin-bottom: 8px;
}

.c3 {
  background: #ffffff;
  border-radius: 12px;
  width: 248px;
  height: 248px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.c4 {
  background: #c6c6c6 height:50px;
  width: 100%;
  border-radius: 0px 0px 12px 12px;
  margin-top: 16px;
  padding-top: 18px;
  padding-left: 50px;
}

<div
  className="container"
>
  <div
    className="title"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        Vous avez
      </div>
      <div
        className="c2"
      >
        ...
      </div>
      <div
        className="c1"
      >
        UserFavorites.content_saved
      </div>
    </div>
  </div>
  <div
    className="cards"
  >
    <div
      className="c3"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div>
        <div
          style={
            {
              "marginLeft": "13px",
              "marginRight": "13px",
            }
          }
        >
          <span
            aria-busy={true}
            aria-live="polite"
          >
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          className="c4"
        />
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div>
        <div
          style={
            {
              "marginLeft": "13px",
              "marginRight": "13px",
            }
          }
        >
          <span
            aria-busy={true}
            aria-live="polite"
          >
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          className="c4"
        />
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        style={
          {
            "padding": "13px",
          }
        }
      >
        <span
          aria-busy={true}
          aria-live="polite"
        >
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
          <span
            className="react-loading-skeleton"
            style={
              {
                "height": "28px",
              }
            }
          >
            ‌
          </span>
          <br />
        </span>
      </div>
      <div>
        <div
          style={
            {
              "marginLeft": "13px",
              "marginRight": "13px",
            }
          }
        >
          <span
            aria-busy={true}
            aria-live="polite"
          >
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
            <span
              className="react-loading-skeleton"
              style={
                {
                  "height": "14px",
                }
              }
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          className="c4"
        />
      </div>
    </div>
  </div>
</div>
`;
